server/protocols/jmap/src/main/java/org/apache/james/jmap/model/MessageFactory.java

```diff
@@ -97,10 +97,10 @@ private String getPreview(IndexableMessage im) {
 
     private String getSubject(IndexableMessage im) {
         return Optional.ofNullable(
-                    Strings.emptyToNull(
                         im.getSubjects()
                             .stream()
-                            .collect(Collectors.joining(MULTIVALUED_HEADERS_SEPARATOR))))
+                            .map(String::trim)
+                            .collect(Collectors.joining(MULTIVALUED_HEADERS_SEPARATOR)))
                 .orElse(NO_SUBJECT);
```

Comment:
in fact after some thinking all the Optional.ofNullable seems useless now that is can't be null.